File: Data Structures & Algorithms/subsets-ii/submission-0.cpp
class Solution {
public:
vector<vector<int>> res;
void backTrack(vector<int>& nums,int i,vector<int> &cur){
     if(i>=nums.size()){
        res.push_back(cur);
        return ;
     }
        cur.push_back(nums[i]);
        backTrack(nums,i+1,cur);
        cur.pop_back();
        while(i+1<nums.size() && nums[i]==nums[i+1]) i++ ;
        backTrack(nums,i+1,cur);
     
}
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
       
        sort(nums.begin(),nums.end());
        vector<int> cur;
        backTrack(nums,0,cur);
        return res;
        
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
       nums.sort()
       n=len(nums)
       res=[]
       for i,a in enumerate(nums):
        if a>0:
            break
        if i>0 and a==nums[i-1]:
            continue
        
        j=i+1
        k=n-1
        while j<k:
            sum=a+nums[j]+nums[k]
            if sum>0:
                k-=1
            elif sum<0:
                j+=1
            else:
                res.append([a,nums[j],nums[k]])
                j+=1
                k-=1
                while j<k and nums[j]==nums[j-1]:
                    j+=1
        
       return res
                

        